nixos/modules/services/cluster/hadoop/hdfs.nix

```diff
@@ -96,15 +120,79 @@ in
         9867 # datanode.ipc.address
       ]);
     })
+    (mkIf cfg.hdfs.journalnode.enable {
+      systemd.services.hdfs-journalnode = {
+        description = "Hadoop HDFS JournalNode";
+        wantedBy = [ "multi-user.target" ];
+        inherit (cfg.hdfs.journalnode) restartIfChanged;
+
+        serviceConfig = {
+          User = "hdfs";
+          SyslogIdentifier = "hdfs-journalnode";
+          ExecStart = "${cfg.package}/bin/hdfs --config ${hadoopConf} journalnode";
+          Restart = "always";
```

Comment:
Might want to add a restart delay?

Reply:
If you're concerned about the services rapidly restarting on failure, the JVM is slow enough to start up that adding a restartdelay here will achieve nothing.